src/Components/firebase-access.jsx

```diff
@@ -3,465 +3,386 @@ import "firebase/auth";
-      currentShares: empArray,
-  }
-  userRef.set(gameInfo);
-
-  for(var i_day = 0; i_day < numDays; i_day++) {
-      var newDoc = userRef.collection('investments').doc(i_day);
-      newDoc.set({
-          change: empArray,
-      });
-  }
-  return userId;
+const debug = (str) => {
 }
 
-// sets up the room to store symbols, indicator graphs, time series,
-// and default portfolio values
-export const setUpRoom = (db,symbolsL,Rounds,password) => {
-  const roomRef = db.collection('Rooms').doc();
-  roomRef.set({
-      day_index: 0,
-      phase: 'no-host',
-      password: password,
-  });
-  const roomID = roomRef.id;
-  initDates(db,symbolsL,Rounds).then((datesD)=> {
-      roomRef.update({
-          symbols: symbolsL,
-          dates: datesD["dates"],
-      });
-      initializeQuiz(symbolsL,roomID,datesD["period"],datesD["dates"]);
-  });
-  return roomID;
-}
+export const addUser = async (roomID,nickname) => {
+    const numDays = (await getDates(roomID)).length;
+    const numSymbols = (await getSymbols(roomID)).length;
+    const roomRef = await getRoomRef(roomID);
+    const userRef = await roomRef.collection('users').doc();
+    const userID = userRef.id;
+    const empArray = Array.from(Array(numSymbols),()=>0);
+    const starting_money = await getStartingMoney(roomID);
 
-export const getUserShares = async (db, roomID, userID, dayIndex) => {
-  const userInvRef = db.collection('Rooms').doc(roomID)
-    .collection('users').doc(userID).collection('investments').doc(dayIndex);
-  const userInvDoc =  await userInvRef.get();
-  const invDocData = userInvDoc.data();
-  return invDocData.change;
-}
+    const gameInfo = {
+        userId: userID,
+        nickname: nickname,
+        net_worth: starting_money,
+        money_left: starting_money,
+        curShares: empArray,
+    }
 
-export const getChartUrl = async (db, roomID, symbol, endDate) => {
-  let images = await db.collection('Rooms').doc(roomID).collection(symbol).doc('images').get();
-  let imagesData = images.data();
-  return imagesData["Stockpublic_image_url"][endDate];
-}
+    userRef.set(gameInfo);
+    for(var i_day = 0; i_day < numDays; i_day++) {
+        await userRef.collection('investments').doc(i_day.toString()).set({
+            change: empArray,
+        });
+    }
 
-export const getTechnicalUrl = async (db, roomID, symbol, endDate) => {
-  let images = await db.collection('Rooms').doc(roomID).collection(symbol).doc('images').get();
-  let imagesData = images.data();
-  return [imagesData["ADXpublic_image_url"][endDate],imagesData["MACDpublic_image_url"][endDate],imagesData['RSIpublic_image_url']]
+    return userID;
 }
 
-function randomDate(start, end) {
-  var date = new Date(+start + Math.random() * (end - start));
-  return date;
+export const changeCash = async (roomID, userID, dayIndex, changeArray, prices) => {
+    const userRef = getUserRef(roomID, userID);
+    const userData = await getUserData(roomID,userID);
```

Comment:
be consistent with spacing
```suggestion
    const userData = await getUserData(roomID, userID);
```